jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -1106,7 +1107,32 @@ static Chunk next(Chunk chunk)
             if (chunk == null)
                 return null;
             if (Content.Chunk.isFailure(chunk))
-                return chunk.isLast() ? chunk : null;
+            {
```

Comment:
I don't think this is the right place to handle this issue, I don't see anything wrong with `Content.Source.read()` always returning an error chunk that contains the exact same exception instance.

I think this should be handled at the place where the exception is thrown, like in `ContentSinkOutputStream`. Most (all?) places should already use `IO.rethrow()` so we can probably come up with some similar static helper, maybe like `Content.Chunk.throwIfFailure(Chunk)`?

We should also add some tests similar to `ContentSinkOutputStreamTest` for each codepath that throws a chunk's exception.

Reply:
@lorban it is now clear that throwing the same instance is not well handled by the JDK and other libraries.

We cannot expect any user-code wrapper (such as `commons-fileupload`) to handle the case where the exception may be self suppressed or there is a loop.

I don't see any other choice than us throwing a different exception every time, or returning a new failure chunk every time.

I'd keep the code we added in #11759 and #13514, but I'm afraid we have to change the read side.

Reply:
@gregw if we do not lose the original cause (following the suggestion @sbordet made in `ExceptionUtil.copyOf()`) then I'm fine with putting the responsibility of making a copy in `Chunk.next()`.

We also have to revisit all our `Content.Source` implementations to make sure they properly call `Chunk.next()`.